report/metadata_template.go

```diff
@@ -75,6 +78,11 @@ func fromCounters(n Node, key string) (string, bool) {
 	return strconv.Itoa(val), ok
 }
 
+func fromSetsIPScope(n Node, key string) (string, bool) {
```

Comment:
This file is fairly generic; is it possible to move the "IP scopes" functionality somewhere more specific to IP addresses or containers?

Reply:
You entier code is fairly generic 😛 
It seems that **ContainerMetadataTemplates** only allows to gather data from one source but there is no option to modify the data. Hence, I cannot extract **IPs** from **IPsWithScope**
MetadataTemplate looks like:
```
type MetadataTemplate struct {
	ID       string  `json:"id"`
	Label    string  `json:"label,omitempty"`    // Human-readable descriptor for this row
	Truncate int     `json:"truncate,omitempty"` // If > 0, truncate the value to this length.
	Datatype string  `json:"dataType,omitempty"`
	Priority float64 `json:"priority,omitempty"`
	From     string  `json:"from,omitempty"` // Defines how to get the value from a report node
}
```
I could add a **To** optional field function, to ask for specific changes on the found data.

`func fromDefault(n Node, key string) (string, bool)` **From** action returns always a string.
My optional **To** field could be a function with `func(data string) string` signature. Then it could be called immediately after the **From** function if it exists.

Reply:
You're right, it is all a bit generic. The original vision was that the app was completely dumb, and all information was supplied by probes.  But this vision didn't really persist so we have a sort of half-and-half situation where some things are heavily coded into the app.

Unfortunately I think the metadata template is passed from the probe, so a function pointer wouldn't work.  Maybe it has to remain the way you had it.